pkg/cmd/kube-router.go

```diff
@@ -76,24 +77,40 @@ func CleanupConfigAndExit() {
 func (kr *KubeRouter) Run() error {
 	var err error
 	var wg sync.WaitGroup
-
 	healthChan := make(chan *healthcheck.ControllerHeartbeat, 10)
 	defer close(healthChan)
-
 	stopCh := make(chan struct{})
 
+	if !(kr.Config.RunFirewall || kr.Config.RunServiceProxy || kr.Config.RunRouter) {
+		glog.Info("Router, Firewall or Service proxy functionality must be specified. Exiting!")
+		os.Exit(0)
+	}
+
 	hc, err := healthcheck.NewHealthController(kr.Config)
 	if err != nil {
 		return errors.New("Failed to create health controller: " + err.Error())
 	}
 	wg.Add(1)
-	go hc.Run(healthChan, stopCh, &wg)
+	go hc.RunServer(stopCh, &wg)
 
-	if !(kr.Config.RunFirewall || kr.Config.RunServiceProxy || kr.Config.RunRouter) {
-		glog.Info("Router, Firewall or Service proxy functionality must be specified. Exiting!")
-		os.Exit(0)
+	informerFactory := informers.NewSharedInformerFactory(kr.Client, 0)
+	svcInformer := informerFactory.Core().V1().Services().Informer()
+	epInformer := informerFactory.Core().V1().Endpoints().Informer()
+	podInformer := informerFactory.Core().V1().Pods().Informer()
+	nodeInformer := informerFactory.Core().V1().Nodes().Informer()
+	nsInformer := informerFactory.Core().V1().Namespaces().Informer()
+	npInformer := informerFactory.Networking().V1().NetworkPolicies().Informer()
+	informerFactory.Start(stopCh)
+
+	err = kr.CacheSyncOrTimeout(informerFactory, stopCh)
+	if err != nil {
+		return errors.New("Failed to synchronize cache: " + err.Error())
 	}
 
```

Comment:
I think we can move 
```
	hc, err := healthcheck.NewHealthController(kr.Config)
	if err != nil {
		return errors.New("Failed to create health controller: " + err.Error())
	}
	wg.Add(1)
	go hc.RunServer(stopCh, &wg)
```

so it doesn't start the health controller until after the pod cache is synced since failing to sync the podcache will halt kube-router and force a restart